### DISCUSSION TITLE
How to run a basic script without imports ?

### DISCUSSION THREAD

**Discussion by bentxt** [leppie/IronScheme#162] [Discussions]
Hi
I work on Macos and I have the problem that I cannot figure out how to run a simple scheme script without imports. Is that even possible?

hello.scm
```
(display "hello")
```

```
$ dotnet IronScheme.ConsoleCore.dll hello.scm
&assertion
&who: expander
&message: "top-level program is missing an (import ---) clause"
&irritants: ()
```

Is it somehow possible to get a basic environment by setting some command line options?

thanks
Ben

**Comment by leppie** [comment]
For an R6RS program it requires imports. 

You can however pass script content to the REPL in the following way:

Windows
```
type hello.scm | IronScheme.Console32-v2.exe
```

or on Linux/MacOS

```
cat hello.scm | dotnet IronScheme.ConsoleCore.dll
```

Or just arb content like (might be slightly different on Linux/MacOS)

```
echo (display "hello")  | IronScheme.Console32-v2.exe
```

**Comment by leppie** [comment]
> Is it somehow possible to get a basic environment by setting some command line options?

There might be something to allow some form of 'preamble', but memory is not serving me well.

Edit:

It seems you can have a file called `init.ss` (in the current working directory) that will be included to the interactive environment (iow REPL).
